components/scream/src/physics/rrtmgp/atmosphere_radiation.cpp

```diff
@@ -1,25 +1,237 @@
+        //  - we can tell RRTMGP it can init all inputs or specify which ones it can init. We call the
+        //    resulting list of inputs the 'initializable' (or initable) inputs. The default is
+        //    that no inputs can be inited.
+        //  - we can request that RRTMGP either inits no inputs or all of the initable ones (as specified
+        //    at the previous point). The default is that RRTMGP must be in charge of init ing ALL or NONE
+        //    of its initable inputs.
+        // Recall that:
+        //  - initable fields may not need initialization (e.g., some other atm proc that
+        //    appears earlier in the atm dag might provide them).
+        std::vector<std::string> rrtmgp_inputs = {
+            "pmid","pint","tmid","tint","col_dry","gas_vmr","sfc_alb_dir","sfc_alb_dif",
+            "mu0","lwp","iwp","rel","rei",
+            "sw_flux_up", "sw_flux_dn", "sw_flux_dn_dir", "lw_flux_up", "lw_flux_dn"
+        };
+        using strvec = std::vector<std::string>;
+        const strvec& allowed_to_init = m_rrtmgp_params.get<strvec>("Initializable Inputs",strvec(0));
+        const bool can_init_all = m_rrtmgp_params.get<bool>("Can Initialize All Inputs", false);
+        const bool init_all_or_none = m_rrtmgp_params.get<bool>("Must Init All Inputs Or None", true);
+        const strvec& initable = can_init_all ? rrtmgp_inputs : allowed_to_init;
+        //const strvec& initable = rrtmgp_inputs;
+        if (initable.size()>0) {
+            bool all_inited = true, all_uninited = true;
+            for (const auto& name : initable) {
+                const auto& f = m_rrtmgp_fields_in.at(name);
+                const auto& track = f.get_header().get_tracking();
+                if (track.get_init_type()==InitType::None) {
+                    // Nobody claimed to init this field. RRTMGPInputsInitializer will take care of it
+                    m_initializer->add_me_as_initializer(f);
+                    all_uninited &= true;
+                    all_inited &= false;
+                } else {
+                    all_uninited &= false;
+                    all_inited &= true;
+                }
+            }
+    
+            // In order to gurantee some consistency between inputs, it is best if RRTMGP
+            // initializes either none or all of the inputs.
+            EKAT_REQUIRE_MSG (
+                !init_all_or_none || all_inited || all_uninited,
+                "Error! Some rrtmgp inputs were marked to be inited by RRTMGP, while others weren't.\n"
+                "       RRTMGP was requested to init either all or none of the inputs.\n"
+            );
+        }
+    }
+    void RRTMGPRadiation::run_impl      (const Real dt) {
+        // Get data from AD; RRTMGP wants YAKL views
+        // TODO: how can I just keep these around without having to create every time?
+        // They are just pointers, so should be able to keep them somewhere else and just associate them once?
```

Comment:
Yes. That's the direction where P3 is heading to. You could have a helper struct like
```
struct YaklArrays {
  using Array2d = yakl::Array<double,2,memDevice,yakl::styleFortran>;

  Array2d pmid;
  Array2d pint;
...
};
```
Then in RRTMGPRadiation you could store a YaklArrays object, and init the views once and for all during `initialize_impl`.

Reply:
I'm going to punt on this for now.

Reply:
Should we make a git-issue so as to not forget about this?  It would probably be tagged as "low-priority" but still.

Reply:
Added a new issue to track. Thanks @AaronDonahue.